packages/near-membrane-base/src/membrane.ts

```diff
@@ -711,9 +741,24 @@ export function createMembraneMarshall(
         let foreignCallableBatchGetPrototypeOfWhenHasNoOwnProperty: CallableBatchGetPrototypeOfWhenHasNoOwnProperty;
         let foreignCallableBatchGetPrototypeOfWhenHasNoOwnPropertyDescriptor: CallableBatchGetPrototypeOfWhenHasNoOwnPropertyDescriptor;
 
+        let fastForeignTargetPointers: WeakSet<Pointer>;
+        let foreignPointerBigInt64ArrayProto: Pointer;
+        let foreignPointerBigUint64ArrayProto: Pointer;
+        let foreignPointerFloat32ArrayProto: Pointer;
+        let foreignPointerFloat64ArrayProto: Pointer;
+        let foreignPointerInt8ArrayProto: Pointer;
+        let foreignPointerInt16ArrayProto: Pointer;
+        let foreignPointerInt32ArrayProto: Pointer;
+        let foreignPointerObjectProto: Pointer;
+        let foreignPointerTypedArrayProto: Pointer;
+        let foreignPointerUint8ArrayProto: Pointer;
+        let foreignPointerUint16ArrayProto: Pointer;
+        let foreignPointerUint32ArrayProto: Pointer;
+        let selectedTarget: undefined | ProxyTarget;
+
+        let useFastPathForTypedArrays = true;
```

Comment:
Typed arrays start out on a faster path for all accesses. If someone modifies its prototypes then the fast path is not used.